metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/lineage.py

```diff
@@ -918,3 +926,178 @@ def test_capability(self, project_id: str) -> None:
     def report_status(self, step: str, status: bool) -> None:
         if self.redundant_run_skip_handler:
             self.redundant_run_skip_handler.report_current_run_status(step, status)
+
+    def gen_lineage_workunits_for_external_table(
+        self,
+        dataset_urn: str,
+        ddl: Optional[str],
+        graph: Optional[DataHubGraph] = None,
+    ) -> Iterable[MetadataWorkUnit]:
+
+        if not ddl:
+            return
+
+        # Expect URIs in `uris=[""]` format
+        uris_match = re.search(r"uris=\[([^\]]+)\]", ddl)
+        if not uris_match:
+            return
+
+        uris_str = uris_match.group(1)
+        source_uris = json.loads(f"[{uris_str}]")
+
+        lineage_info = self.get_lineage_for_external_table(
+            dataset_urn=dataset_urn,
+            source_uris=source_uris,
+            graph=graph,
+        )
+
+        if lineage_info:
+            yield MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn, aspect=lineage_info
+            ).as_workunit()
+
+    def get_lineage_for_external_table(
+        self,
+        dataset_urn: str,
+        source_uris: List[str],
+        graph: Optional[DataHubGraph] = None,
+    ) -> Optional[UpstreamLineageClass]:
+
+        upstreams_list: List[UpstreamClass] = []
+        fine_grained_lineages: List[FineGrainedLineageClass] = []
+        gcs_urns: Set[str] = set()
+
+        for source_uri in source_uris:
+            # Check that storage_location have the gs:// prefix.
+            # Right now we are only supporting GCS lineage
+            if not gcs_utils.is_gcs_uri(source_uri):
+                continue
+            gcs_path = self._get_gcs_path(source_uri)
+
+            if gcs_path is None:
```

Comment:
should we log this and or collect it in Report?

Reply:
I have already added into report, you can find it in self._get_gcs_path(source_uri) method 
```
    def _get_gcs_path(self, path: str) -> Optional[str]:
        if self.config.gcs_lineage_config:
            for path_spec in self.config.gcs_lineage_config.path_specs:
                if not path_spec.allowed(path):
                    logger.debug(
                        f"Skipping gcs path {path} as it does not match any path spec."
                    )
                    self.report.num_lineage_dropped_gcs_path += 1                    
                    ...
```